pkg/platform/aks/aks.go

```diff
@@ -0,0 +1,354 @@
+			Detail: fmt.Sprintf("%q field is empty and %q environment variable "+
+				"is not defined. At least one of these should be defined",
+				"SubscriptionID", subscriptionIDEnv),
+		})
+	}
+
+	if c.TenantID == "" && os.Getenv(tenantIDEnv) == "" {
+		d = append(d, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "cannot find the Azure client ID",
+			Detail: fmt.Sprintf("%q field is empty and %q environment variable "+
+				"is not defined. At least one of these should be defined", "TenantID", tenantIDEnv),
+		})
+	}
+
+	f := map[string]string{
+		"AssetDir":          c.AssetDir,
+		"ClusterName":       c.ClusterName,
+		"ResourceGroupName": c.ResourceGroupName,
+	}
+
+	for k, v := range f {
+		if v == "" {
+			d = append(d, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  fmt.Sprintf("field %q can't be empty", k),
+			})
+		}
+	}
+
+	return d
+}
+
+// checkCredentials checks if credentials are correctly defined.
+func (c *config) checkCredentials() hcl.Diagnostics {
+	var d hcl.Diagnostics
+
+	// If the application name is defined, we assume, that we work as highly privileged
+	// account, which has permissions to create new Azure AD application, so Client ID
+	// and Client Secret fields are not needed.
+	if c.ApplicationName != "" {
+		if c.ClientID != "" {
+			d = append(d, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "ClientID and ApplicationName are mutually exclusive",
+			})
+		}
+
+		if c.ClientSecret != "" {
+			d = append(d, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "ClientSecret and ApplicationName are mutually exclusive",
+			})
+		}
+
+		return d
+	}
+
+	if c.ClientSecret == "" && os.Getenv(clientSecretEnv) == "" {
+		d = append(d, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "cannot find the Azure client secret",
+			Detail: fmt.Sprintf("%q field is empty and %q environment variable "+
+				"is not defined. At least one of these should be defined", "ClientSecret", clientSecretEnv),
+		})
+	}
+
+	if c.ClientID == "" && os.Getenv(clientIDEnv) == "" {
+		d = append(d, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "cannot find the Azure client ID",
+			Detail: fmt.Sprintf("%q field is empty and %q environment variable is "+
+				"not defined. At least one of these should be defined", "ClientID", clientIDEnv),
+		})
+	}
+
+	return d
+}
+
+// checkNotEmptyWorkers checks if the cluster has at least 1 node pool defined.
+func (c *config) checkNotEmptyWorkers() hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	if len(c.WorkerPools) == 0 {
+		diagnostics = append(diagnostics, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "At least one worker pool must be defined",
+			Detail:   "Make sure to define at least one worker pool block in your cluster block",
+		})
+	}
+
+	return diagnostics
+}
+
+// checkWorkerPoolNamesUnique verifies that all worker pool names are unique.
+func (c *config) checkWorkerPoolNamesUnique() hcl.Diagnostics {
```

Comment:
I smell copy&paste here ;)

I guess the idea is to refactor it later?

Reply:
I guess so, but I'm not sure if it's worth it for now. Worker pool struct would need to implement the interface for getting the names etc. seems like an over-engineering a bit :smile:

Reply:
If others are okay adding tech debt to have this feature soon, as we have planned a refactor for this, I'm fine.